python/hail/expr/functions.py

```diff
@@ -3756,3 +3756,42 @@ def liftover(x, dest_reference_genome, min_match=0.95):
         Use 'add_liftover' to load a liftover chain file.""".format(rg.name, dest_reference_genome.name))
 
     return _func(method_name, rtype, to_expr(dest_reference_genome.name, tstr), x, to_expr(min_match, tfloat))
+
+
+@typecheck(f=func_spec(1, expr_float64),
+           min=expr_float64,
+           max=expr_float64)
+def uniroot(f: Callable, min, max):
+    """Finds a root of the function `f` within the interval from `min` to `max`.
+
+    Examples
+    --------
+
+    >>> hl.uniroot(lambda x: x - 1, -5, 5).value
+    1.0
+
+    Notes
+    -----
+    `f(min)` and `f(max)` must have opposite signs.
```

Comment:
Not true, they can be zero. more correct to say "must not have the same sign."

I don't see this condition checked anywhere, add it inside the scala code? In R, results in an error message:
```
Error in uniroot(f1, c(0, 10)) : 
  f() values at end points not of opposite sign
```

Reply:
it's inside the scala code, yeah.

Reply:
Where?